python/hail/expr/aggregators.py

```diff
@@ -670,7 +670,7 @@ def filter(condition, expr):
 
     Parameters
     ----------
-    condition : :class:`hail.expr.expression.BooleanExpression` or function
+    condition : :class:`hail.expr.expression.BooleanExpression` or 1-argument function
```

Comment:
Is there a way to concisely say the function return type must be boolean? I see you check the result is boolean below.

Reply:
oooo, good idea. We can do `func_spec(1, expr_bool)`. Great idea!